### ISSUE TITLE
newsfeed banner

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#1327]
at top of try something random box (or ??) - add content from Meta/Global

**Comment by dustymc** [comment]
I had to rewrite the header model to fit this in

PLEASE TEST THE LAYOUT AT http://arctos-test.tacc.utexas.edu/

Default:

<img width="1385" alt="screen shot 2017-11-14 at 11 58 57 am" src="https://user-images.githubusercontent.com/5720791/32801905-49866b16-c933-11e7-927a-016429316994.png">

Hover:

<img width="1385" alt="screen shot 2017-11-14 at 11 59 05 am" src="https://user-images.githubusercontent.com/5720791/32801916-4e40487a-c933-11e7-8fa5-0e9364d012c2.png">

**Comment by campmlc** [comment]
Yes, looks great. Will this be Twitter feed?

**Comment by dustymc** [comment]
Thanks. No, it's not twitter, it's a new field in manage/meta/global.

<img width="909" alt="screen shot 2017-11-14 at 3 04 51 pm" src="https://user-images.githubusercontent.com/5720791/32809806-314f945e-c94d-11e7-850d-2fb5d6e05b96.png">

I don't think it should be twitter either - I'm not sure who our twitter audience is, but it doesn't include me and probably many other Arctos users. The "newsfeed" is targeted at active Arctos users. I don't think there's reason to assume overlap between those groups.

**Comment by campmlc** [comment]
I agree. I'm not on Twitter either. However, it was proposed as a way that
the AWG could have access to post news issues without having to modify the
web portal directly.

**Comment by dustymc** [comment]
This doesn't require CODE being modified, only a form entry, and everyone on the AWG likely already has access to that form. (I think it's manage_global rights.) I'm not sure if that's what you're saying or not, and I'm always up for better ideas.

**Comment by campmlc** [comment]
Do we have documentation yet on how to use the form? Would be a good thing
to disseminate.

**Comment by dustymc** [comment]
Want to write that and I'll answer questions? I'm not sure what needs documenting??

**Comment by Jegelewicz** [comment]
What needs documenting:
"everyone on the AWG likely already has access to that form. (I think it's manage_global rights.)"
I don't know anything about managing global rights or forms that let me make changes to the website - I didn't even know I had access to do such things. Maybe more than documentation is required. I feel like I need some sort of training to be an AWG member....

**Comment by campmlc** [comment]
I feel the same as Teresa. I guess we need to put together some guidelines.

**Comment by dustymc** [comment]
No disagreement for any of that, and i'm happy to do whatever, I just don't think I'm the best person to write docs - I don't generally need documentation to understand my code, and so I'm never quite sure what about the code might need documented.

"manage_global" (actually "global_admin") is a role.

<img width="314" alt="screen shot 2017-11-15 at 12 27 28 pm" src="https://user-images.githubusercontent.com/5720791/32858457-6941051a-ca00-11e7-972a-261fac4b5dfc.png">

It works like any other role, and can be accessed from....

<img width="340" alt="screen shot 2017-11-15 at 12 28 11 pm" src="https://user-images.githubusercontent.com/5720791/32858498-898eabd8-ca00-11e7-8347-f9bb8f339895.png">

or from editAgent.

<img width="364" alt="screen shot 2017-11-15 at 12 29 06 pm" src="https://user-images.githubusercontent.com/5720791/32858523-9b4c3426-ca00-11e7-8be3-8bdf765b37b4.png">

then

"Def"

<img width="1374" alt="screen shot 2017-11-15 at 12 29 37 pm" src="https://user-images.githubusercontent.com/5720791/32858537-aea1aff6-ca00-11e7-8529-06ccfc6208ca.png">

provides a CRUD (https://en.wikipedia.org/wiki/CRUD) definition.

In addition to table-level access control, roles provide or restrict access to forms. You can see that here...

<img width="351" alt="screen shot 2017-11-15 at 12 31 21 pm" src="https://user-images.githubusercontent.com/5720791/32858634-f8fdda7a-ca00-11e7-8710-7bed920c6b24.png">

or set it here

<img width="356" alt="screen shot 2017-11-15 at 12 32 06 pm" src="https://user-images.githubusercontent.com/5720791/32858648-0635920a-ca01-11e7-82c9-ee61fac7fa5f.png">





<img width="346" alt="screen shot 2017-11-15 at 12 20 52 pm" src="https://user-images.githubusercontent.com/5720791/32858172-72765df2-c9ff-11e7-96fc-2da14824fd12.png">


controls

* newsfeed banner
* Google Enterprise credentials
* Google Analytics credentials
* EZID (our source of DOIs) credentials
* GenBank credentials
* Who gets various emails
* IPs which are immune to blacklisting

**Comment by campmlc** [comment]
Under definition of the role, then, we need to add "forms permissions" with
the following added and defined. Also, we need to add documentation to
Github how to actually edit these forms and manage credentials etc. Who is
currently doing these things who can help with this documentation?

controls

   - newsfeed banner
   - Google Enterprise credentials
   - Google Analytics credentials
   - EZID (our source of DOIs) credentials
   - GenBank credentials
   - Who gets various emails
   - IPs which are immune to blacklisting

**Comment by dustymc** [comment]
I am not sure what you're talking about.

**Comment by dustymc** [comment]
The tool is done+in production. Seems there may still be documentation issues that need resolved so I'll temporarily leave this open.

**Comment by mkoo** [comment]
There should be some guidelines on content too-- this is NEWS that will be
on the public front page of Arctos for all collections, right? (GLOBAL
management)

simply and nicely done, Dusty!

**Comment by dustymc** [comment]
It is on ALL pages for ALL collections.

Thanks!

**Comment by ekrimmel** [comment]
If I understand the status of this issue correctly I *should* be seeing the newsfeed banner on arctos.database.museum, yes? That isn't what I'm seeing. If I am not logged in nothing is changed from pre-newsfeed. If I am logged in there is not only not a newsfeed but the "try something random" section is also gone. Both logged in and not logged in our institutional header images don't show up like they used to on catalog record pages... I was looking at this in Chrome and Firefox

**Comment by dustymc** [comment]
I don't think there's any content in the newsfeed yet.

> the "try something random" section is also gone.

Did you turn it off? My Stuff (https://arctos.database.museum/myArctos.cfm), ....

<img width="219" alt="screen shot 2017-11-17 at 9 59 47 am" src="https://user-images.githubusercontent.com/5720791/32961610-14bc2b44-cb7e-11e7-815e-c2b435b7fdd8.png">

I'm still seeing TSR, but I did muck with CSS so it's possible yours is hiding or something. Screenshot and resolution details, please.

>institutional header images don't show up

Woops! I'll patch that back in ASAP.

**Comment by ekrimmel** [comment]
Oops, yep, I had turned off TSR. And thanks for explaining that if there's no content then no content will show up... derrr!

I guess my next question is that when I view my own permissions I have the global_admin role assigned, but under Manage Data-->Metadata I only see Encumbrances and Manage Collection, nothing about Global Settings. Should I?

**Comment by dustymc** [comment]
Ah - it seems to also require manage_codetables

<img width="569" alt="screen shot 2017-11-17 at 10 33 40 am" src="https://user-images.githubusercontent.com/5720791/32962854-da5a6cb8-cb82-11e7-8c5a-b32b04605b1a.png">

I'm ALWAYS up for guidance regarding permissions - maybe everyone on the AWG should have code table access, or that form should be more open, or whatever.

**Comment by jtgiermakowski** [comment]
Howdy,

Liking the newsfeed but as Erika pointed out "Both logged in and not logged in, our institutional header images don't show up like they used to on catalog record pages"

i tried on Chrome and Firefox as well as for different collections (MSB:Herp, MSB:Mamm)

the HEADER_COLOR works fine but HEADER_IMAGE is not happening. Also missing is 
COLLECTION_URL | COLLECTION_LINK_TEXT
INSTITUTION_URL | INSTITUTION_LINK_TEXT

**Comment by dustymc** [comment]
The collection header switcher-inner should be switching now.

**Comment by ccicero** [comment]
Sorry, just chiming in now. I like how the news banner expands and can link out, but the addition of the banner has screwed up the placement of the logo relative to the text "Collaborative Collection Management Solution" - can those be lined up better? Also, who is going to be responsible for news announcements (our Communications officer?) and what are the guidelines for announcements? Seems like we need to have some guidelines before we put this into production. If we have a news feed, then we need to keep it active and dynamic.

**Comment by dustymc** [comment]
>screwed up the placement of the logo

Screenshot please.

**Comment by campmlc** [comment]
Maybe it should auto expire and vanish a certain amount of time after
posting?

**Comment by ccicero** [comment]
![image](https://user-images.githubusercontent.com/967275/33143993-81dc6bd0-cf70-11e7-90fd-c23cc9c48871.png)

versus

![image](https://user-images.githubusercontent.com/967275/33144015-954f207c-cf70-11e7-8eae-f64e7990c7fb.png)

Auto-expire/vanish is a good idea if that's possible. We just don't want stale news, and it takes work to keep this fresh.

**Comment by dustymc** [comment]
Derp - thanks! Fixed and patched into prod.

**Comment by dustymc** [comment]
>Auto-expire

Yes, it's possible. When would news expire? 2 weeks?

**Comment by ccicero** [comment]
I just updated the news item in test but it's not updating on the banner - cached? 

Could we choose an expiration date (add a calendar widget)? Some news may require a shorter and known expiration - e.g., planned outages - whereas we may want to keep others posted longer.

**Comment by dustymc** [comment]
>cached

Yup.

<img width="192" alt="screen shot 2017-11-22 at 11 14 09 am" src="https://user-images.githubusercontent.com/5720791/33145714-5d4e3cf2-cf76-11e7-9c7c-ce2b3fd5c9d4.png">

But I just turned that off in test so it should be live now.

>choose an expiration date 

Good idea, I like.

**Comment by dustymc** [comment]
>choose an expiration date

Done at test, for next release.

<img width="459" alt="screen shot 2017-11-22 at 12 18 06 pm" src="https://user-images.githubusercontent.com/5720791/33148070-396d38e8-cf7f-11e7-9122-071706ce98a8.png">

**Comment by dustymc** [comment]
Needs live forced-refresh preview

**Comment by dustymc** [comment]
save now flushes cache

**Comment by campmlc** [comment]
Hi all, I just went in to the banner page to fix the dates from "5/27 to 5/27" to "5/27 to 5/29".  I like that I can no just type free text and don't have to deal with encoding the English. But it is very concerning to be making these edits on a page where there is so much other critical stuff. I don't have a cat, but if I did and he walked across my keyboard while I turned away for a sip of coffee, I could really mess up something in there while working on the banner. Any way to move the banner to its own form in PG? Maybe with separate permissions, for communications only?
